app/services/instrument_offline_reservation_canceler.rb

```diff
@@ -0,0 +1,44 @@
+class InstrumentOfflineReservationCanceler
+
+  def cancel!
+    reservations_to_cancel.each do |reservation|
+      reservation.transaction do
+        cancel_reservation(reservation)
+        Notifier.offline_cancellation_notification(reservation)
+      end
+    end
+  end
+
+  private
+
+  def admin
+    # OrderDetail#cancel_reservation needs an object that responds to #id
+    @admin ||= OpenStruct.new(id: 0)
+  end
+
+  def cancel_reservation(reservation)
+    reservation
+      .order_detail
+      .cancel_reservation(admin, OrderStatus.canceled.first, true, false)
```

Comment:
I was thinking it might be nice to move the canceled_reason into this method call, but that would make the signature really gross without doing some extra refactoring elsewhere.